### PR TITLE
fix(clerk-js): Delete user profile image

### PR DESCRIPTION
**Pull request by raptisj**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected
 
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Delete user profile image

### CODE REVIEW

File: packages/clerk-js/src/core/resources/User.ts [hunk 1]
```diff
@@ -209,6 +209,10 @@ export class User extends BaseResource implements UserResource {
 
   setProfileImage = (params: SetProfileImageParams): Promise<ImageResource> => {
     const { file } = params || {};
+
+    if (!file) {
```
Comment (gkats) [inline packages/clerk-js/src/core/resources/User.ts:213]: :question: Is this check enough? Should we explicitly check that the `file` parameter is set to `null` before we trigger a DELETE request?

**Review by SokratisVidros — APPROVED** [review]

File: packages/clerk-js/src/ui/localization/defaultEnglishResource.ts [hunk 2]
```diff
@@ -275,6 +275,7 @@ export const defaultResource: DeepRequired<LocalizationResource> = {
       title: 'Update profile',
       imageFormTitle: 'Profile image',
       imageFormSubtitle: 'Upload image',
+      destructiveActionSubtitle: 'Delete image',
```
Comment (gkats) [inline packages/clerk-js/src/ui/localization/defaultEnglishResource.ts:278]: :question: Should this say "Delete" or "Remove"?

Comment (SokratisVidros) [reply packages/clerk-js/src/ui/localization/defaultEnglishResource.ts:278]: Remove sounds better

File: packages/clerk-js/src/core/resources/User.ts (hunk 1 above)
Comment (raptisj) [reply packages/clerk-js/src/core/resources/User.ts:213]: @gkats what edge case do you have in mind?

File: packages/clerk-js/src/ui/localization/defaultEnglishResource.ts (hunk 2 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/localization/defaultEnglishResource.ts:278]: @SokratisVidros , @nikosdouvlis what do you think?

Comment (desiprisg) [reply packages/clerk-js/src/ui/localization/defaultEnglishResource.ts:278]: Since the image can't be retrieved, I'd say delete is better here.

File: packages/clerk-js/src/core/resources/User.ts (hunk 1 above)
Comment (gkats) [reply packages/clerk-js/src/core/resources/User.ts:213]: I don't have an edge case in mind. It's just an API choice.

```js
// No file in the parameters, is this intentional?
user.setProfileImage();

// vs

// File is passed and set to null
user.setProfileImage({ file: null });
```

I'm not sure how TS will behave with the above, but in plain JS, undefined and null will be different.

cc @SokratisVidros

Comment (desiprisg) [reply packages/clerk-js/src/core/resources/User.ts:213]: Typescript should complain but not everyone will be using typescript, so maybe it should indeed be explicit.

Comment (nikosdouvlis) [reply packages/clerk-js/src/core/resources/User.ts]: We need to be explicit about this, otherwise any improper call invocation would delete the image. Lets check for `=== null`

File: packages/clerk-js/src/core/resources/User.ts [hunk 3]
```diff
@@ -209,6 +209,10 @@ export class User extends BaseResource implements UserResource {
 
   setProfileImage = (params: SetProfileImageParams): Promise<ImageResource> => {
     const { file } = params || {};
+
+    if (file === null) {
+      return Image.delete(`${this.path()}/profile_image`);
+    }
     return Image.create(`${this.path()}/profile_image`, {
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/core/resources/User.ts:216]: Not part of this PR, but its a little unfortunate that the user knows about the image endpoint 🙃

File: packages/clerk-js/src/ui/components/UserProfile/UserProfileAvatarUploader.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ export const UserProfileAvatarUploader = (
     <AvatarUploader
       {...rest}
       title={localizationKeys('userProfile.profilePage.imageFormTitle')}
+      hasImageUrl={!new RegExp('gravatar').test(user.profileImageUrl || '')}
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/UserProfile/UserProfileAvatarUploader.tsx:14]: Would this be enough?
```suggestion
      hasImageUrl={(user.profileImageUrl || '').includes('gravatar')}
```

File: packages/clerk-js/src/ui/elements/AvatarUploader.tsx [hunk 5]
```diff
@@ -8,7 +8,8 @@ import { FileDropArea } from './FileDropArea';
 export type AvatarUploaderProps = {
   title: LocalizationKey;
   avatarPreview: React.ReactElement;
-  onAvatarChange: (file: File) => Promise<unknown>;
+  onAvatarChange: (file: File | null) => Promise<unknown>;
+  hasImageUrl?: boolean;
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/elements/AvatarUploader.tsx:12]: I think that the `hasImageUrl` prop doesn't really communicate its intent. What do you think about replacing this with an `onAvatarRemove?: () => Promise<unknown>` handler instead?

File: packages/clerk-js/src/ui/elements/AvatarUploader.tsx [hunk 6]
```diff
@@ -70,6 +72,18 @@ export const AvatarUploader = (props: AvatarUploaderProps) => {
...
             />
+
+            {hasImageUrl && !showUpload && (
+              <Button
+                localizationKey={localizationKeys('userProfile.profilePage.destructiveActionSubtitle')}
+                isDisabled={card.isLoading}
+                sx={t => ({ color: t.colors.$danger500 })}
+                variant='link'
+                onClick={() => {
+                  void onAvatarChange(null);
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/elements/AvatarUploader.tsx:83]: Lets also disable the Delete button while the request runs

File: packages/types/src/localization.ts [hunk 7]
```diff
@@ -282,6 +282,7 @@ type _LocalizationResource = {
       title: LocalizationValue;
       imageFormTitle: LocalizationValue;
       imageFormSubtitle: LocalizationValue;
+      destructiveActionSubtitle: LocalizationValue;
```
Comment (nikosdouvlis) [inline packages/types/src/localization.ts:285]: Do you think we can group the delete button key with the rest, by prefixing it with `imageForm`? eg: 
```suggestion
      imageFormDestructiveAction: LocalizationValue;
```

File: packages/clerk-js/src/core/resources/User.ts (hunk 1 above)
Comment (SokratisVidros) [reply packages/clerk-js/src/core/resources/User.ts]: ☝️

File: packages/clerk-js/src/ui/localization/defaultEnglishResource.ts (hunk 2 above)
Comment (SokratisVidros) [reply packages/clerk-js/src/ui/localization/defaultEnglishResource.ts:278]: Let's do Remove

File: packages/clerk-js/src/core/resources/User.ts (hunk 3 above)
Comment (SokratisVidros) [reply packages/clerk-js/src/core/resources/User.ts:216]: At least it's nested 😌

File: packages/clerk-js/src/ui/elements/AvatarUploader.tsx (hunk 5 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/elements/AvatarUploader.tsx:12]: @nikosdouvlis we want to check if avatar is the defualt gravatar one or a custom one. Maybe `hasDefaultImage`? I don't get how `onAvatarRemove` would work in this case

File: packages/types/src/localization.ts (hunk 7 above)
Comment (raptisj) [reply packages/types/src/localization.ts:285]: @nikosdouvlis probably `imageFormDestructiveActionSubtitle` is better since it matches the convention

File: packages/clerk-js/src/ui/components/UserProfile/UserProfileAvatarUploader.tsx (hunk 4 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/UserProfile/UserProfileAvatarUploader.tsx:14]: @nikosdouvlis yes this will do 🙃

**Review by SokratisVidros — APPROVED** [review]
